contrib/src/main/java/com/datatorrent/contrib/enrich/DelimitedFSLoader.java

```diff
@@ -0,0 +1,189 @@
+   * @param hasHeader
+   *          If set to true, the first line will be skipped
+   */
+  public DelimitedFSLoader(String fieldDescription, boolean hasHeader)
+  {
+    this(fieldDescription);
+    this.hasHeader = hasHeader;
+  }
+
+  /**
+   * 
+   * @param fieldDescription
+   *          String representing fields and its type in the same order as it
+   *          appears in the records. E.g.
+   *          empId:INTEGER,address:STRING,phone:LONG Supported data types are
+   *          BOOLEAN, INTEGER, FLOAT, DOUBLE, LONG, SHORT, CHARACTER, STRING If
+   *          header is present, field names must match the header.
+   * @param hasHeader
+   *          If set to true, header will be skipped
+   * @param delimiter
+   */
+  public DelimitedFSLoader(String fieldDescription, boolean hasHeader, char delimiter)
+  {
+    this(fieldDescription, hasHeader);
+    strategy = new CSVStrategy(delimiter, CSVStrategy.ENCAPSULATOR_DISABLED, CSVStrategy.COMMENTS_DISABLED);
+  }
+
+  /**
+   * 
+   * @param fieldDescription
+   *          String representing fields and its type in the same order as it
+   *          appears in the records. E.g.
+   *          empId:INTEGER,address:STRING,phone:LONG Supported data types are
+   *          BOOLEAN, INTEGER, FLOAT, DOUBLE, LONG, SHORT, CHARACTER, STRING
+   * @param hasHeader
+   *          If set to true, the first line will be skipped
+   * @param delimiter
+   * @param quote
+   */
+  public DelimitedFSLoader(String fieldDescription, boolean hasHeader, char delimiter, char quote)
+  {
+    this(fieldDescription, hasHeader);
+    strategy = new CSVStrategy(delimiter, quote, CSVStrategy.COMMENTS_DISABLED);
+  }
+
+  /**
+   * Extracts the fields from a delimited record and returns a map containing
+   * field names and values
+   */
+  @Override
+  Map<String, Object> extractFields(String line)
+  {
+    try {
+      String[] values = new CSVParser(new StringReader(line), strategy).getLine();
+      if (hasHeader && Arrays.deepEquals(values, header)) {
+        return null;
+      }
+      Map<String, Object> map = Maps.newHashMap();
+      int i = 0;
+      for (Field field : fields) {
+        map.put(field.getName(), getValue(field.getType(), values[i++]));
+      }
+      return map;
+    } catch (IOException e) {
+      logger.error("Exception while extracting fields {}", e);
+    }
+    return null;
+  }
+
+  private Object getValue(FIELD_TYPE type, String value)
+  {
+    switch (type) {
+      case BOOLEAN:
+        return Boolean.parseBoolean(value);
+      case DOUBLE:
+        return Double.parseDouble(value);
+      case INTEGER:
+        return Integer.parseInt(value);
+      case FLOAT:
+        return Float.parseFloat(value);
+      case LONG:
+        return Long.parseLong(value);
+      case SHORT:
+        return Short.parseShort(value);
+      case CHARACTER:
+        return value.charAt(0);
+      default:
+        return value;
```

Comment:
What about DATE FIELD_TYPE? I think you need to parse that too. Also which date format is allowed?